### PR TITLE
New App: Visualizer Coffee

### PR DESCRIPTION
**Pull request by jeffbean**
Adding a first version of an app that will display espresso shot information based on the data found from visualizer.coffee.

The main information is number of shots today, and when the last shot was.

# Description
Adds the application displaying information that is found in the coffee API to share with others and maybe yourself on how much coffee you may or may not be having in a day :D

visualizercoffee

# Copilot

copilot:all

### CODE REVIEW

**Comment by jeffbean** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by jeffbean** [comment]
recheck

**Review by mokolodi1 — COMMENTED** [review]
Non-blocking comments after a quick read-through

File: apps/visualizercoffee/visualizercoffee.star [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+        render.Marquee(
+            width = 64,
+            child = render.Text(msg),
+            align = "center",
+        ),
+    )
+
+def get_my_shots(auth_token):
+    # https://visualizer.coffee/api/shots/?page=1&items=10
+    resp = http.get(
```
[the remark this answers is not in the document]
Comment (jeffbean) [reply apps/visualizercoffee/visualizercoffee.star:116]: thanks, updated

**Comment by mokolodi1** [comment]
LGTM

**Review by mokolodi1 — COMMENTED** [review]
LGTM!

**Comment by matslina** [comment]
Sorry for the terrible delay on this one. We're doing a lot of work on Tidbyt Gen 2 right now, and the community repo suffers as a consequence. =(

We usually try to set up Tidbyt-owned credentials for oauth apps, but I can't seem to find anything on visualizer.coffee on how to set those up. Do you by any chance have any pointers?

**Comment by jeffbean** [comment]
> Sorry for the terrible delay on this one. We're doing a lot of work on Tidbyt Gen 2 right now, and the community repo suffers as a consequence. =(
> 
> We usually try to set up Tidbyt-owned credentials for oauth apps, but I can't seem to find anything on visualizer.coffee on how to set those up. Do you by any chance have any pointers?

Ah, so the app does OAuth but requires contacting them to enable it for an account. https://github.com/miharekar/visualizer?tab=readme-ov-file#api

If this is something of a blocker I can always swap to basic auth, but OAuth would be ideal.

**Comment by matslina** [comment]
There's some sort of OAuth issue going on. I'm troubleshooting and will follow up.

**Comment by matslina** [comment]
Haven't been able to get past it. Authorization codes arrive as they should, but the access_token endpoint is returning 404 on POST.

**Comment by jeffbean** [comment]
hmm, ok. ill do some debugging.

thanks for setting it up!

**Comment by matslina** [comment]
Feel free to shoot me a DM on discord if I can help. I've been trying to get this working for a bit now, but failing so far.

**Comment by matslina** [comment]
Hopefully resolved in #2353

**Pull request merged**